src/lib.rs

```diff
@@ -689,33 +737,31 @@ decl_module! {
                             LegStatus::ExecutionToBeSkipped(receipt.signer.clone(), receipt.receipt_uid)
                         );
                     } else {
-                        match T::Asset::unsafe_increase_custody_allowance(
-                            did,
-                            leg_details.asset,
-                            did,
-                            SettlementDID.as_id(),
-                            leg_details.amount
-                        ) {
-                            Ok(_) => <InstructionLegStatus<T>>::insert(instruction_id, leg_id, LegStatus::ExecutionPending),
-                            Err(_) => return Rollback(Err(Error::<T>::FailedToTakeCustodialOwnership))
+                        if T::Portfolio::lock_tokens(&leg_details.from, &leg_details.asset, &leg_details.amount).is_err() {
```

Comment:
Why we are locking tokens if someone has not provided the receipt? Shouldn't it need explicit authorization ?

Reply:
Someone might want to authorize an instruction with 10 legs but provide receipts for only 2 legs.

Reply:
Need to add code comments for clarification